### DISCUSSION TITLE
RuntimeWarning: invalid value encountered in sqrt

### DISCUSSION THREAD

**Discussion by JunejaAmmy** [CURENT/andes#305] [Q&A]
Hi Dr. Cui,

I'm facing below issue while running TDS. Can you please help me?

I have followed below steps with use case: **ieee14_dgprct.xlsx**
1. Ran the simulation for 1500 seconds with tstep = 0.33 and shrinkt = 0
2. Altered the value of `tm0` of GENROU_1 to 1.55
3. Continued the simulation to 1800 seconds and faced below issue.

```
C:\Users\AmrinderS\.andes\pycode\ESST3A.py:21: RuntimeWarning: invalid value encountered in sqrt
  return ((-v + vbus, LAW2_y*VB_y*ue - vout, -UEL + UEL0, ue*(-IN*VE + KC*XadIfd), -FEX_y + select([less_equal(IN, 0),less_equal(IN, 0.433),less_equal(IN, 0.75),less_equal(IN, 1),greater(IN, 1),__trues], [__ones,1 - 0.577*IN,sqrt(0.75 - IN**2),1.732 - 1.732*IN,__zeros,__zeros], default=nan), FEX_y*VE - VB_x, VBMAX*VB_lim_zu + VB_lim_zi*VB_x - VB_y, KG*vout - VG_x, VGMAX*VG_lim_zu + VG_lim_zi*VG_x - VG_y, LAW1_y - VG_y - vrs, -vref + vref0, -LG_y - vi + vref, HLI_zi*vi + HLI_zl*VIMIN + HLI_zu*VIMAX - vil, HG_lt_z0*UEL + HG_lt_z1*vil - HG_y, LL_LT1_z1*LL_LT2_z1*(-LL_x + LL_y) + LL_x*TB - LL_y*TB + TC*(HG_y - LL_x), ue*(-vf0 + vout), 0, 0, 0, 0, 0, 0, 0))
Time step reduced to zero. Convergence is not likely.
Simulation terminated at t=1502.6400 s.
```

**Comment by cuihantao** [comment]
You disabled automatic step size shrinking. The numerical integration could not converge after some disturbance.

What you can do is enable `shrinkt` and see if it converged. If yes, you can manually interpolate the data at needed timestamps.

**Comment by cuihantao** [comment]
Altering `tm0` of generators will trigger electromechanical transient response that is unlikely to solve using a large step size of 0.33 s.
